server/worker/worker.py

```diff
@@ -1,21 +1,48 @@
+import signal
+import sys
 import time
 
 from server.database import db_session
-from server.worker.tasks import run_new_tasks
+from server.worker.tasks import (
+    claim_next_task,
+    reset_task,
+    run_task,
+)
 from server.sentry import configure_sentry
 from server.websession import cleanup_sessions
 
 # We have to import all of the api so that the modules load and all of the task
 # handlers get registered as background_tasks.
 from server import api  # pylint: disable=unused-import
 
-if __name__ == "__main__":
-    configure_sentry()
+
+def run_worker():
+    task = None
+
+    # If we're interrupted in the middle of a task, reset it before exiting so
+    # it can be picked up by another worker.
+    def interrupt_handler(*_args):
+        nonlocal task
+        if task:
+            reset_task(task)
+            task = None
+        sys.exit(1)
+
+    signal.signal(signal.SIGTERM, interrupt_handler)
+    signal.signal(signal.SIGINT, interrupt_handler)
```

Comment:
this looks good in principle, have we tested that these are the signals we get on Heroku and we have time to try to reset the task?

Reply:
Haven't tested it yet, just going on the [Heroku docs](https://devcenter.heroku.com/articles/dyno-shutdown-behavior) so far. They claim the dyno will receive one or more SIGTERMs, then get 30s to shutdown before receiving a SIGKILL.